compiler/src/main/java/com/squareup/anvil/compiler/codegen/KotlinPoet.kt

```diff
@@ -62,12 +62,6 @@ internal fun ClassDescriptor.asClassName(): ClassName =
   )
 
 internal fun FqName.asClassName(module: ModuleDescriptor): ClassName {
-  try {
-    return ClassName.bestGuess(asString())
-  } catch (ignored: IllegalArgumentException) {
-    // Probably lowercase class. Try to resolve the class below.
-  }
```

Comment:
I need to benchmark this in our repository before merging. The code below tries to resolve the class, which is more overhead and could slow down compile times.